tests/envs/utils.py

```diff
@@ -0,0 +1,78 @@
+"""Finds all the specs that we can test with"""
+from typing import Optional
+
+import numpy as np
+
+import gym
+from gym import logger
+from gym.envs.registration import EnvSpec
+
+
+def try_make_env(env_spec: EnvSpec) -> Optional[gym.Env]:
+    """Tries to make the environment showing if it is possible. Warning no wrappers, including time limit."""
+    try:
+        return env_spec.make(disable_env_checker=True).unwrapped
+    except ImportError as e:
+        logger.warn(f"Not testing {env_spec.id} due to error: {e}")
+        return None
+
+
+# Tries to make all environment to previous recreation if not necessary.
+all_testing_initialised_envs = list(
+    filter(None, [try_make_env(env_spec) for env_spec in gym.envs.registry.values()])
+)
+
+# All testing, mujoco and gym environment specs
+all_testing_env_specs = [env.spec for env in all_testing_initialised_envs]
+mujoco_testing_env_specs = [
+    env_spec
+    for env_spec in all_testing_env_specs
+    if "gym.envs.mujoco" in env_spec.entry_point
+]
+gym_testing_env_specs = [
+    env_spec
+    for env_spec in all_testing_env_specs
+    if any(
+        f"gym.{ep}" in env_spec.entry_point
+        for ep in ["box2d", "classic_control", "toy_text"]
+    )
+]
+# TODO, add minimum testing env spec in testing
```

Comment:
How will this be used? Can it be added in a later PR if this isn't used anywhere now?

Reply:
Yes, my plan was that to be a future pr. 
The idea was with more environments being added in the future, phys3d, phys2d, etc
This is going to continue slow down CI but I don't believe that every test needs to check over all environments
Therefore, we can have a reduce list of representative environments, i.e., a box2d environment, a classic control environment, etc
This would be useful for wrappers where we shouldn't need to check the wrapper works with every environments, we just need to check that the wrapper works with each type of observation, action space